pmd-core/src/main/java/net/sourceforge/pmd/lang/AntlrTokenManager.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang;
+
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+
+/**
+ * Generic token manager implementation for all Antlr lexers.
+ */
+public class AntlrTokenManager implements TokenManager {
+    private final Lexer lexer;
+    private String fileName;
+
+    /**
+     * Constructor
+     *
+     * @param lexer The lexer
+     * @param fileName The file name
+     */
+    public AntlrTokenManager(final Lexer lexer, final String fileName) {
+        this.lexer = lexer;
+        this.fileName = fileName;
+    }
+
+    @Override
+    public Object getNextToken() {
+        return lexer.nextToken();
+    }
+
+    @Override
+    public void setFileName(String fileName) {
```

Comment:
this method should get called at some point for lexical errors to produce proper messages

Reply:
In the current implementation we are keeping the current implementation on PMD, that took the `fileName` from `SourceCode`, the only thing we did here was moving the fileName inside `AntlrTokenManager` and provide a setter to support other scenarios

Reply:
Instead of making the error using `sourceCode.getFileName()` we use [`tokenManager.getFileName()`](https://github.com/pmd/pmd/pull/1309/files#diff-281c2cec2c4bead1c7f6f0177378fecfR34)